### ISSUE TITLE
Add support for host tools on the `arm64e-apple-darwin` target

### ISSUE THREAD

**Issue by arttet** [rust-lang/rust#132945]
**Known Issues**:
- [ ] Integrate `jemallocator` for the `arm64e-apple-darwin` target.
       Blocked by https://github.com/tikv/jemallocator/issues/102

**Definition of Done**:
- [ ] The `arm64e-apple-darwin` target meets the requirements of the [Tier 1 Target Policy](https://doc.rust-lang.org/nightly/rustc/target-tier-policy.html#tier-1-target-policy).

See https://github.com/rust-lang/rust/issues/73628
Duplicate of https://github.com/rust-lang/rust/issues/131884
